pygmt/tests/test_grdproject.py

```diff
@@ -0,0 +1,59 @@
+"""
+Tests for grdproject.
+"""
+import os
+
+import numpy.testing as npt
+import pytest
+from pygmt import grdinfo, grdproject
+from pygmt.datasets import load_earth_relief
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import GMTTempFile
+
+
+@pytest.fixture(scope="module", name="grid")
+def fixture_grid():
+    """
+    Load the grid data from the sample earth_relief file.
+    """
+    return load_earth_relief(resolution="01d", region=[-5, 5, -5, 5])
+
+
+def test_grdproject_file_out(grid):
+    """
+    grdproject with an outgrid set.
+    """
+    with GMTTempFile(suffix=".nc") as tmpfile:
+        result = grdproject(grid=grid, projection="M10c", outgrid=tmpfile.name)
+        assert result is None  # return value is None
+        assert os.path.exists(path=tmpfile.name)  # check that outgrid exists
+        result = grdinfo(tmpfile.name, per_column=True).strip().split()
+        npt.assert_approx_equal(float(result[0]), 0)  # x min
+        npt.assert_approx_equal(float(result[1]), 10)  # x max
+        npt.assert_approx_equal(float(result[2]), -5.55111512313e-17)  # y min
+        npt.assert_approx_equal(float(result[3]), 9.94585661273)  # y max
+        npt.assert_approx_equal(float(result[4]), -5130.48193359)  # min
+        npt.assert_approx_equal(float(result[5]), -152.585281372)  # max
```

Comment:
This test is going to break we refactor `grdinfo` to return non-string outputs (as mentioned in https://github.com/GenericMappingTools/pygmt/issues/593#issuecomment-895144705). I'm debating whether we should check the statistics by loading the output grid into `xarray`, something like so:

https://github.com/GenericMappingTools/pygmt/blob/3f4090e60c12ba4b40e8b757ef1f199a04b1ff1a/pygmt/tests/test_triangulate.py#L97-L99